maintenance/wikia/renameUser.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/**
+ * Perform user rename process
+ *
+ * @author Macbre
+ * @ingroup Maintenance
+ */
+
+putenv( 'SERVER_ID=177' ); // run in the context of community wiki
+
+require_once( dirname( __FILE__ ) . '/../Maintenance.php' );
+
+/**
+ * Maintenance script class
+ */
+class RenameUserMaintenance extends Maintenance {
+
+	/**
+	 * Set script options
+	 */
+	public function __construct() {
+		parent::__construct();
+
+		$this->addOption( 'old-username', 'User to rename', true );
+		$this->addOption( 'new-username', 'New name for the user', true );
+		$this->addOption( 'reason', 'The rename reason' );
+
+		$this->mDescription = 'This script performs user rename from the command line';
+	}
+	/**
+	 * Script entry point
+	 */
+	public function execute() {
+		global $wgUser;
+		$wgUser = User::newFromName( 'WikiaBot' );
+
+		// get options
+		$oldName = $this->getOption( 'old-username' );
+		$newName = $this->getOption( 'new-username' );
+		$reason = $this->getOption( 'reason', 'User rename' );
+
+		// set up the logger
+		$logger = Wikia\Logger\WikiaLogger::instance();
+		$logger->pushContext( [
+			'old-name' => $oldName,
+			'new-name' => $newName,
+			'reason'   => $reason
+		] );
+
+		$logger->info( __CLASS__ . ': start' );
+
+		$renameProcess = new RenameUserProcess( $oldName, $newName, true, $reason );
+		$res = $renameProcess->run();
```

Comment:
AFAIK 3 years ago (it might've changed in the meantime though) the main task spawns more tasks, one for each wiki the user edited so after this line the process is not yet completed.